src/py42/services/_connection.py

```diff
@@ -104,6 +104,9 @@ def from_microservice_prefix(cls, connection, prefix, auth=None, session=None):
     def host_address(self):
         return self._get_host_address()
 
+    def add_header(self, header):
```

Comment:
Sorry to make you rework things, but I don't think this should be needed, see my discussion on how to approach needing the extra header by implementing this in `V3Auth`.

Reply:
Makes sense! I've updated V3Auth to add the header directly to the request, and removed the add_header method from Connection